### PR TITLE
Initial Commit for Babelfish v2.6

### PR DESCRIPTION
**Pull request by hash-16**
### Description

- Update Github action scripts
- Create upgrade test folder for last minor version.
- Bump Babelfish version string in babelfish_version.h

### Issues Resolved

[List any issues this PR will resolve]

### Test Scenarios Covered ###
* **Use case based -**

* **Boundary conditions -**

* **Arbitrary inputs -**

* **Negative test cases -**

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/babelfish_version.h [hunk 1]
```diff
@@ -8,7 +8,7 @@
  *-------------------------------------------------------------------------
  */
 
-#define BABELFISH_VERSION_STR "2.5.0"
-#define BABELFISH_INTERNAL_VERSION_STR "Babelfish 14.7.0.0"
+#define BABELFISH_VERSION_STR "2.6.0"
+#define BABELFISH_INTERNAL_VERSION_STR "Babelfish 14.8.0.0"
```
Comment (rishabhtanwar29) [inline contrib/babelfishpg_tsql/src/babelfish_version.h]: Should be 14.9.0.0

File: .github/template/version-branch-template.yml [hunk 2]
```diff
@@ -25,6 +25,9 @@
 '14.6':
   engine_branch: BABEL_2_3_STABLE__PG_14_6
   extension_branch: BABEL_2_3_STABLE
+'14.8'
```
Comment (rishabhtanwar29) [inline .github/template/version-branch-template.yml]: Missing `:`

Comment (hash-16) [reply .github/template/version-branch-template.yml]: Addressed

File: contrib/babelfishpg_tsql/src/babelfish_version.h (hunk 1 above)
Comment (hash-16) [reply contrib/babelfishpg_tsql/src/babelfish_version.h]: Addressed

File: contrib/babelfishpg_tsql/src/babelfish_version.h [hunk 3]
```diff
@@ -8,7 +8,7 @@
  *-------------------------------------------------------------------------
  */
 
-#define BABELFISH_VERSION_STR "2.5.0"
-#define BABELFISH_INTERNAL_VERSION_STR "Babelfish 14.7.0.0"
+#define BABELFISH_VERSION_STR "2.7.0"
```
Comment (rishabhtanwar29) [inline contrib/babelfishpg_tsql/src/babelfish_version.h]: BABELFISH_VERSION_STR should be 2.6.0

File: .github/configuration/upgrade-test-configuration.yml [hunk 4]
```diff
@@ -38,6 +38,16 @@ upgrade-version: [[
...
+[
+  {
+    version: 14.8,
+    upgrade-type: null
+  },
+  {
+    version: target.latest,
+    upgrade-type: minor
+  }
+],
```
Comment (rishabhtanwar29) [inline .github/configuration/upgrade-test-configuration.yml]: Let's not add a new workflow, instead update the above entry (14.6 -> latest) to 14.8 -> latest.

**Review by rishabhtanwar29 — APPROVED** [review]

**Review by shardgupta — APPROVED** [review]

**Pull request merged**